power-pay-backend/src/main/java/com/adorsys/gis/powerpay/powerpaybackend/services/SendMoneyImpl.java

```diff
@@ -1,8 +1,55 @@
 package com.adorsys.gis.powerpay.powerpaybackend.services;
-
+import com.adorsys.gis.powerpay.powerpaybackend.domain.ProcedureStatus;
+import com.adorsys.gis.powerpay.powerpaybackend.domain.Transaction;
+import com.adorsys.gis.powerpay.powerpaybackend.errorhandling.InsufficientFundsException;
+import com.adorsys.gis.powerpay.powerpaybackend.repository.MoneyTransferRepository;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.TransactionException;
 
 @Service
 public class SendMoneyImpl implements SendMoney{
 
+    @Autowired
+    private MoneyTransferRepository moneyTranferRepository;
+
+    private Transaction transaction;
+
+    @Override
+    public Transaction send(String phoneNumber, String receiverPhoneNumber, Double amount, String currency, Integer id)
+            throws InsufficientFundsException, TransactionException {
+        if (phoneNumber == null) {
+            throw new IllegalArgumentException("Phone number cannot be null.");
+        }
+        if (amount <= 0) {
+            throw new IllegalArgumentException("Transfer amount must be positive.");
+        }
+
+        if (!hasSufficientFunds(phoneNumber, amount)) {
+            throw new InsufficientFundsException("Insufficient funds for transfer.");
+        }
+
+        try {
+            if (transaction != null) {
```

Comment:
...and create the transaction object right before here

Reply:
Sir when we do that we get this error relating to the moneyTransferRepository object when running the test for the SendMoneyImpl class:

```
com.adorsys.gis.powerpay.powerpaybackend.services.SendMoneyImpl$1: Failed to save transaction: Cannot invoke "com.adorsys.gis.powerpay.powerpaybackend.repository.MoneyTransferRepository.save(Object)" because "this.moneyTransferRepository" is null
Caused by: java.lang.NullPointerException: Cannot invoke "com.adorsys.gis.powerpay.powerpaybackend.repository.MoneyTransferRepository.save(Object)" because "this.moneyTransferRepository" is null
```

whereas the test works just fine when the transaction object is created where it is right now.

Reply:
Hello @stephane-segning ,
The test is failing because, the object moneyTransferRepository is not being injected properly.
If you check the code, you will find that I did it like:

```
@Service
public class SendMoneyImpl implements SendMoney{

    @Autowired
    private MoneyTransferRepository moneyTransferRepository;
    public SendMoneyImpl (MoneyTransferRepository moneyTrans
```

I have also tried this approach:

```
@Service
public class SendMoneyImpl implements SendMoney{

    private final MoneyTransferRepository moneyTransferRepository;

    public SendMoneyImpl (MoneyTransferRepository moneyTransferRepository){
        this.moneyTransferRepository = moneyTransferRepository;
    }
```
None has solved the issue for us. 
The error remains the same:
**### ```
com.adorsys.gis.powerpay.powerpaybackend.services.SendMoneyImpl$1: Failed to save transaction: Cannot invoke "com.adorsys.gis.powerpay.powerpaybackend.repository.MoneyTransferRepository.save(Object)" because "this.moneyTransferRepository" is null
 at com.adorsys.gis.powerpay.powerpaybackend.services.SendMoneyImpl.send([SendMoneyImpl.java:50]
```**
We will appreciate some insights so we can move forward with this.
Thank you.

Reply:
You'll have this kind of error when the IoC of Spring is not working, typically in test cases.

Reply:
Because `SendMoney` is the interface and `SendMoneyImpl` the implementation using the `@Service` annotation, then you can write your test using `@MockBean`, as descriibe [here](https://www.baeldung.com/spring-boot-testing).

Reply:
The test has now been implemented with the ```@MockBean``` sir. Thanks for the guidance.